File: src/main/java/io/dustin/finstage/financial/dto/SymbolListRequest.java
package io.dustin.finstage.financial.dto;

import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor
public class SymbolListRequest {

    @Min(1)
    private int page = 1;

    @Min(1)
    @Max(100)
    private int size = 20;

    public static SymbolListRequest of(int page, int size) {
        SymbolListRequest request = new SymbolListRequest();
        request.page = page;
        request.size = size;
        return request;
    }
}

File: src/main/kotlin/io/dustin/finstage/financial/dto/SymbolListRequest.kt
package io.dustin.finstage.financial.dto

import jakarta.validation.constraints.Max
import jakarta.validation.constraints.Min

data class SymbolListRequest(

        @field:Min(1)
        val page: Int = 1,

        @field:Min(1)
        @field:Max(100)
        val size: Int = 20
) {
    companion object {
        fun of(page: Int, size: Int): SymbolListRequest {
            return SymbolListRequest(page, size)
        }
    }
}
